### PR TITLE
Return error message if user tapped "Don't Allow"

### PR DESCRIPTION
**Pull request by extempl**
Based on https://stackoverflow.com/a/34189489/506954

### CODE REVIEW

**Comment by macdonst** [comment]
@extempl I've been testing this but I can't get it to fail when the user clicks "Don't Allow" the code never seems to get called. Can you give me a better reproduction scenario?

**Comment by extempl** [comment]
@macdonst That was a bit obsolete version, I've updated it to how it works on my end now:
Added two events `allowed` and `disallowed`. For checking access it uses same approach as used in `hasPermission` method. 
Also added ability to `off` all handlers of certain event, as `allowed` will fire each time app is resumed, so, on my end, I am listening for `allowed`/`disallowed` event only once before the first register. For `resume` I using built-in `hasPermissions` method.

Probably the code is need in polishing, I am not an ios-dev.

File: src/ios/PushPlugin.m [hunk 1]
```diff
@@ -282,6 +282,28 @@ - (void)init:(CDVInvokedUrlCommand*)command;
...
+                                                                                                    queue:[NSOperationQueue mainQueue]
+                                                                                               usingBlock:^(NSNotification * _Nonnull note) {
+               [NSThread sleepForTimeInterval:0.1f];
+               BOOL enabled = NO;
+               id<UIApplicationDelegate> appDelegate = [UIApplication sharedApplication].delegate;
+               if ([appDelegate respondsToSelector:@selector(userHasRemoteNotificationsEnabled)]) {
+                   enabled = [appDelegate performSelector:@selector(userHasRemoteNotificationsEnabled)];
+               }
+               if (enabled) {
+                   NSLog(@"User allowed Notifications for the App");
```
Comment (macdonst) [inline src/ios/PushPlugin.m]: I'm cool with the log but I think we can lose everything else in this if.

File: www/push.js [hunk 2]
```diff
@@ -34,7 +34,9 @@ var PushNotification = function () {
     this.handlers = {
       registration: [],
       notification: [],
-      error: []
+      error: [],
+      allowed: [],
```
Comment (macdonst) [inline www/push.js]: Let's get rid of the allowed/disallowed event listeners.

File: www/push.js [hunk 3]
```diff
@@ -47,7 +49,9 @@ var PushNotification = function () {
 
     // triggered on registration and notification
     var success = function success(result) {
-      if (result && typeof result.registrationId !== 'undefined') {
+      if (result === 'allowed') {
```
Comment (macdonst) [inline www/push.js]: Scratch this.

File: src/ios/PushPlugin.m [hunk 4]
```diff
@@ -282,6 +282,28 @@ - (void)init:(CDVInvokedUrlCommand*)command;
...
+                   CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:@"allowed"];
+                   [pluginResult setKeepCallbackAsBool:YES];
+                   [self.commandDelegate sendPluginResult:pluginResult callbackId:self.callbackId];
+               }
+               else {
+                   NSLog(@"User disallowed Notifications for the App");
+                   [self failWithMessage:self.callbackId withMsg:@"disallowed" withError:Nil];
```
Comment (macdonst) [inline src/ios/PushPlugin.m]: Maybe just give a more detailed error message here.

File: www/push.js [hunk 5]
```diff
@@ -77,6 +81,9 @@ var PushNotification = function () {
 
     // triggered on error
     var fail = function fail(msg) {
+      if (msg === 'disallowed') {
```
Comment (macdonst) [inline www/push.js]: Probably just get rid of this and let the regular error handler take care of it.

**Pull request closed without merging**